fuse-products/src/main/java/software/tnb/product/integration/generator/IntegrationGenerator.java

```diff
@@ -95,6 +97,14 @@ public static void processCustomizers(AbstractIntegrationBuilder<?> integrationB
             new BytemanCustomizer()
         );
 
+        // the http server port customizer should be added by default, however, if there is any other instance of that class already added
+        // (by using .port(port, boolean) method)
+        if (integrationBuilder.getCustomizers().stream().noneMatch(c -> c instanceof HTTPServerPortCustomizer)) {
+            integrationBuilder.addCustomizer(new HTTPServerPortCustomizer(true));
+        }
+
+        Collections.reverse(integrationBuilder.getCustomizers());
```

Comment:
why? :thinking:

Reply:
because the user-defined customizers should take precedence over the default ones, but currently if the user added a customizer in the test, it was invoked first

Reply:
probably there is currently no such case, but I think it makes sense :grinning:

Reply:
sure I assumed it would be something like that, I would just add a comment why it's happening :D